crates/storage/db/src/abstraction/table.rs

```diff
@@ -101,19 +101,17 @@ pub trait TableImporter<'tx>: for<'a> DbTxMut<'a> {
         source_tx: &R,
         from: Option<<T as Table>::Key>,
         to: <T as Table>::Key,
-    ) -> Result<(), Error> {
+    ) -> Result<(), Error>
+    where
+        T::Key: Default,
+    {
         let mut destination_cursor = self.cursor_write::<T>()?;
         let mut source_cursor = source_tx.cursor_read::<T>()?;
 
-        for row in source_cursor.walk(from)? {
+        let from = from.unwrap_or_default();
+        for row in source_cursor.walk_range(from..to)? {
```

Comment:
internally, both `walk` and `walk_range` would call `MDBX_SET_RANGE`, which would position a cursor at the first key greater than or equal to the specified key. here we rely on `<T::KEY>::default()` being a zero value, which might not always be true
```suggestion
        let source_range = match from {
            Some(from) => source_cursor.walk_range(from..=to),
            None => source_cursor.walk_range(..=to),
        };
        for row in source_range? {
```